### ISSUE TITLE
Iterative process for finding lateral genes

### ISSUE THREAD

**Issue by LoaiNaom** [tanaylab/metacells#79]
Hi @orenbenkiki !

I am following the vignette for running MC2. I chose the initial lateral genes (e.g. `BASE_LATERAL_GENE_NAMES` and `BASE_LATERAL_GENE_PATTERNS`).

After that I ran the process of similarity between genes and grouping them into modules. Then I plot the heatmaps, and according to them, I choose the new lateral genes (the genes or the modules with *). Then, I run again the function `update_lateral_genes` to add the new lateral genes.

so this is the first round. According to the vignette you used MCView, which I was not able to run for some reason. Do you think I can run the second iteration just the same way I did the first time ?

Meaning that right now I have the new lateral genes, after I did the first iteration. I can run the similarity process again, this should give me new lateral genes, right ?
If not, what other way would you recommend to continue this lateral genes process, since MCView did not work for me.

**Comment by orenbenkiki** [comment]
Yes, you can run everything without using MCView. However it is extremely useful to use MCView to look at the data, especially heatmaps and gene-gene plots. If you have trouble running MCView, I suggest you contact aviezer.lifshitz@weizmann.ac.il or better yet open a ticket in https://github.com/tanaylab/MCView

Edit: We don't typically do a 2nd round of related genes using correlations and so on. This is OK for a 1st pass but we found that for more accurate analysis we need more fine-grained tools. We therefore instead look at the resulting metacells in more detail (typically using MCView).

**Comment by LoaiNaom** [comment]
Thanks @orenbenkiki !

I am also trying to run this on spatial data which has much less genes (about 1000 genes). The correlation matrix between modules gives me an all blue heatmap, which indicated all values are constant. And I am not getting any correlated genes inside those modules. Do you think that this algorithm is fit for spatial data ? or perhaps spatial needs a special treatment unlike scRNA-seq data ?

Thanks!

**Comment by orenbenkiki** [comment]
What do you mean by "spatial data"? Is it still a collection of cells, with the UMI counts of each gene for each cell, but with some additional per-cell spatial metadata? If so, then what you see sounds unreasonable; the expression of different genes in different cells would not be that highly correlated with each other.

**Comment by LoaiNaom** [comment]
@orenbenkiki it's MERFISH data, there might a lot of contamination between cells due to the segmentation. Our gene modules are empty when running `sb.clustermap`. This is the output I am getting : 

![image](https://github.com/user-attachments/assets/1cabdc8f-5961-4462-823e-99df191c1559)

**Comment by orenbenkiki** [comment]
I would be highly suspicious of this data... how many cells do you have there?

**Comment by LoaiNaom** [comment]
@orenbenkiki We have 500,000 cells and 1000 genes. We are using the raw counts data.

**Comment by orenbenkiki** [comment]
It did find ~31 modules, but this level of uniform correlation between them is highly suspicious. I would do some sanity check by looking at the gene modules and searching for a case where two strong genes known to be uncorrelated or anti-correlated (from knowledge of the biology) and computing their correlation in the data, just to make sure there's nothing fishy going one.

You can also metacells with your original lateral genes list and explore further. Again, heatmap of marker genes and gene-gene plots are very powerful tools for understanding the data.

**Comment by LoaiNaom** [comment]
@orenbenkiki it worked out. Thank you for your help, honestly very helpful ! I'm almost done

So now my metacells data is normalized (the `metacells.X` is basically fractions of each gene). Do I just go on with my analysis just as I would do with scRNA-seq data ? (e.g., perform UMAPs, find DEGs, run statistical tests, etc.)

Also - when running UMAPs, no further normalization of `metacells.X` is needed ?

**Comment by orenbenkiki** [comment]
UMAP is pretty generic and doesn't require (in general) normalization. Other tools may or may not require converting the gene fractions to UMIs, but this must be done with care (because 1 - you need to pick some arbitrary number of total UMIs per metacell and 2 - you end up with non-integer UMIs counts so you need to do some stochastic rounding or something). We find that it is better to look at the fractions themselves as they make it easier to compare results between data sets.
